internal/testrunner/runners/system/runner.go

```diff
@@ -201,60 +402,83 @@ func (r *runner) newResult(name string) *testrunner.ResultComposer {
-		return result.WithError(errors.New("missing Elasticsearch client"))
+		return errors.New("missing Elasticsearch client")
 	}
 	if r.options.KibanaClient == nil {
-		return result.WithError(errors.New("missing Kibana client"))
+		return errors.New("missing Kibana client")
 	}
 
-	stackVersion, err := r.options.KibanaClient.Version()
+	r.stackVersion, err = r.options.KibanaClient.Version()
 	if err != nil {
-		return result.WithError(fmt.Errorf("cannot request Kibana version: %w", err))
+		return fmt.Errorf("cannot request Kibana version: %w", err)
 	}
 
 	devDeployPath, err := servicedeployer.FindDevDeployPath(servicedeployer.FactoryOptions{
-		Profile:            r.options.Profile,
 		PackageRootPath:    r.options.PackageRootPath,
-		DataStreamRootPath: dataStreamPath,
+		DataStreamRootPath: r.dataStreamPath,
 		DevDeployDir:       DevDeployDir,
-		StackVersion:       stackVersion.Version(),
 	})
 	if err != nil {
-		return result.WithError(fmt.Errorf("_dev/deploy directory not found: %w", err))
+		return fmt.Errorf("_dev/deploy directory not found: %w", err)
 	}
 
-	cfgFiles, err := listConfigFiles(r.options.TestFolder.Path)
-	if err != nil {
-		return result.WithError(fmt.Errorf("failed listing test case config cfgFiles: %w", err))
+	if r.options.ConfigFilePath != "" {
+		allCfgFiles, err := listConfigFiles(filepath.Dir(r.options.ConfigFilePath))
+		if err != nil {
+			return fmt.Errorf("failed listing test case config cfgFiles: %w", err)
+		}
+		baseFile := filepath.Base(r.options.ConfigFilePath)
+		for _, cfg := range allCfgFiles {
+			if cfg == baseFile {
+				r.cfgFiles = append(r.cfgFiles, baseFile)
+			}
+		}
+	} else {
+		r.cfgFiles, err = listConfigFiles(r.options.TestFolder.Path)
+		if err != nil {
+			return fmt.Errorf("failed listing test case config cfgFiles: %w", err)
+		}
 	}
 
 	variantsFile, err := servicedeployer.ReadVariantsFile(devDeployPath)
 	if err != nil && !errors.Is(err, os.ErrNotExist) {
-		return result.WithError(fmt.Errorf("can't read service variant: %w", err))
+		return fmt.Errorf("can't read service variant: %w", err)
+	}
+
+	r.variants = r.selectVariants(variantsFile)
+
+	return nil
+}
+
+func (r *runner) run() (results []testrunner.TestResult, err error) {
+	result := r.newResult("(init)")
+	if err = r.initRun(); err != nil {
+		return result.WithError(err)
+	}
+
+	if _, err = os.Stat(r.locationManager.ServiceSetupDir()); !os.IsNotExist(err) {
+		return result.WithError(fmt.Errorf("failed to run tests, required to tear down previous setup run: %s exists", r.locationManager.ServiceSetupDir()))
 	}
 
 	startTesting := time.Now()
-	for _, cfgFile := range cfgFiles {
-		for _, variantName := range r.selectVariants(variantsFile) {
-			partial, err := r.runTestPerVariant(result, locationManager, cfgFile, dataStreamPath, variantName, stackVersion.Version())
+	for _, cfgFile := range r.cfgFiles {
+		for _, variantName := range r.variants {
+			partial, err := r.runTestPerVariant(result, cfgFile, variantName)
```

Comment:
Yes, that could be helpful, as you mention, to reduce the time and resources spent in system tests.

Not sure if there would be more changes needed, at least it would be needed to ensure that for every test the (old) documents present in the data stream are deleted before running each one.

Reply:
> for every test the (old) documents present in the data stream are deleted before running each one.

Yes, this should be done independently of the running service.